### PR TITLE
EPE-389 Fix net_plugin stall during head catchup

### PR DESCRIPTION
**Pull request by rusty-block-one**
This fix handles situations where a node is in head_catchup state and
recieves the sync_last_requested_num and then does a sync_update_expected
for the next block. And the next block has already been received by the
node. In this instance, the block will not receive the next block and
will be effectively hung.

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [x] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: plugins/net_plugin/net_plugin.cpp [hunk 1]
```diff
@@ -1965,6 +1965,9 @@ namespace eosio {
 
          if( set_state_to_head_catchup ) {
             set_state( head_catchup );
+            if( blk_num == sync_last_requested_num ) {
+                request_next_chunk( std::move( g_sync ));
```
Comment (heifner) [inline plugins/net_plugin/net_plugin.cpp:1969]: I have not looked at the logic of this, but you do need to lock `g_sync` before the call. `request_next_chunk` assumes `g_sync` is locked. It was unlocked on line 1945 above.

Comment (heifner) [reply plugins/net_plugin/net_plugin.cpp:1970]: `g_sync` needs to be locked before accessing `sync_last_requested_num` as that var is protected by `sync_mtx`.

Comment (rusty-block-one) [reply plugins/net_plugin/net_plugin.cpp]: killing this pull request and creating a new one

**Comment by rusty-block-one** [comment]
Reissuing the pull request

**Pull request closed without merging**